plugins/database/cassandra/tls.go

```diff
@@ -0,0 +1,118 @@
+package cassandra
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+
+	"github.com/hashicorp/vault/sdk/helper/certutil"
+	"github.com/hashicorp/vault/sdk/helper/errutil"
+)
+
+func jsonBundleToTLSConfig(rawJSON string, tlsMinVersion uint16, serverName string, insecureSkipVerify bool) (*tls.Config, error) {
+	certBundle := certutil.CertBundle{}
+	err := json.Unmarshal([]byte(rawJSON), &certBundle)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse JSON: %w", err)
+	}
+
+	if certBundle.IssuingCA != "" && len(certBundle.CAChain) > 0 {
+		return nil, fmt.Errorf("issuing_ca and ca_chain cannot both be specified")
+	}
+	if certBundle.IssuingCA != "" {
+		certBundle.CAChain = []string{certBundle.IssuingCA}
+		certBundle.IssuingCA = ""
+	}
+
+	return toClientTLSConfig(certBundle.Certificate, certBundle.PrivateKey, certBundle.CAChain, tlsMinVersion, serverName, insecureSkipVerify)
+}
+
+func pemBundleToTLSConfig(pemBundle string, tlsMinVersion uint16, serverName string, insecureSkipVerify bool) (*tls.Config, error) {
+	if len(pemBundle) == 0 {
+		return nil, errutil.UserError{Err: "empty pem bundle"}
+	}
+
+	pemBytes := []byte(pemBundle)
+	var pemBlock *pem.Block
+
+	certificate := ""
+	privateKey := ""
+	caChain := []string{}
+
+	count := 0
+	for len(pemBytes) > 0 {
```

Comment:
Does the ordering while parsing matter? For example, should we check/ensure that there's no private key being inserted in the middle of a chain?

Reply:
As far as I can tell there is no required ordering down stream. The current `ParsePEMBundle` function also doesn't enforce any ordering (other than the implicit assertion that the client certificate is before the CA cert). We can enforce ordering if we want, but the current implementation doesn't require it.

Reply:
I noticed that `ParsePEMBundle` performs validity checks around the client certs and the CA chain, [via](https://github.com/hashicorp/vault/blob/4e0fa83bcda67ea3548d73f4016bbe5a03f3119e/sdk/helper/certutil/helpers.go#L187) `parsedBundle.Verify()`. Wondering if we should do the same.

Reply:
I looked into the `Verify` function and it is specific to mTLS so we can't do the exact same validation. I'm suspicious that we don't need to do much of what it's doing in that function as it's probably handled downstream in the TLS code, but I'll try to double check that.